.github/workflows/tag_release.yml

```diff
@@ -0,0 +1,74 @@
+name: Create Release and Attach Files
+
+on:
+  pull_request:
+    types:
+      - closed
+    branches:
+      - main
+
+jobs:
+  create_release:
+    if: ${{ github.event.pull_request.merged == true }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Determine new version
+        run: |
+          # Extract the latest version number from Changelog.md
+          VERSION=$(grep -Po '(?<=\[)\d+\.\d+\.\d+(?=\])' CHANGELOG.md | head -1)
+          echo "Current version: $VERSION"
+          echo "APP_VERSION=$VERSION" >> $GITHUB_ENV
+
+      - name: Zip NetFramework folders
+        run: |
+          zip -r NetFramework_4.0_Any.zip NetFramework_4.0_Any/
+          zip -r NetFramework_4.5_Any.zip NetFramework_4.5_Any/
+          zip -r NetFramework_4.7_Any.zip NetFramework_4.7_Any/
+
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: v${{ env.APP_VERSION }}
+          release_name: Release v${{ env.APP_VERSION }}
+          body: Release of version ${{ env.APP_VERSION }}
+          draft: false
+          prerelease: false
+
+      - name: Upload Release Asset for NetFramework 4.0
```

Comment:
What's the benefit of the zip files here? Since a release automatically contains the source code tar.gz, it seems like duplicate assets

Reply:
It is a duplicate, but typically you are only going to need one set of those so it makes it easy to just grab the necessary ones. I can remove it if you think that's dumb.